fix(compiler): Fix look up of entryComponents in AOT Summaries generated by references from router configuration.

AlabasterAxe opened the pull request:
Previously, when you attempted to bootstrap a component that had a router-outlet using ngsummaries, it would complain that the component was not provided by any module even if it was. This commit fixes a mistake (AFAICT) which caused the lookup of the component in the AOT summaries to fail.

I believe this change is safe. I've run the Angular tests and affected tests within Google and there were no breakages caused by this change.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
When you attempt to bootstrap a component that includes a routerconfig using AOT Summaries, the test fails complaining that the Component that you referenced in the router config was not included in any modules even though it is.

"Error: Component AotRouterChildCmp is not part of any NgModule or the module has not been imported into your module."

Issue Number: N/A


## What is the new behavior?
You are able to bootstrap the component successfully.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information
The formatting changes were generated by clang formatter.

AlabasterAxe commented:
Looking into the test failures...

AlabasterAxe commented:
I looked into the failures and they don't look related (they don't mention the files that I created or modified). I tried to replicate the failures locally but they failed to build on my mac with "ld: unknown option: -no-as-needed". Any advice here?

vicb commented:
http://test/OCL:204768985:BASE:205516175:1532181675178:fdbe0eca

jasonaden reviewed (approved):
Tests look good from my side. @alexhub should take a look on the compiler side.

alxhub commented on packages/tsconfig-metadata.json:
```diff
@@ -1,23 +1,23 @@
 {
   "extends": "./tsconfig.json",
-
```
Were these formatting changes caused by a particular tool? Please revert as they seem to have nothing to do with your PR.

AlabasterAxe replied:
Hi, yeah I had reformat on save turned on in VS Code and the changes seemed uncontroversial so I left them in. Removed!

alxhub reviewed (changes requested):
Thank you very much - this looks like a needed change to me!

I'll request 2 things:

1) Revert the unneeded tsconfig.json formatting changes.

2) Squash your commits into a single commit with a meaningful commit message, explaining what the problem was and how the fix works.

AlabasterAxe commented:
Thanks for the reviews!
@alxhub: re squashing: I want to make sure I'm doing the right thing because my familiarity with git is pretty low. Should I rebase my branch and force push? I believe in other GitHub projects I worked on there was a way to squash at merge time but I'm not seeing that option here.

- Matt

AlabasterAxe committed:
fix(compiler): Fix look up of entryComponents in AOT Summaries

Previously, when you attempted to bootstrap a component that had a
router-outlet using ngsummaries, it would complain that the component
was not provided by any module even if it was. This commit fixes a
mistake (AFAICT) which caused the lookup of the component in the AOT
summaries to fail.

I believe this change is safe. I've run the affected tests within Google
and there were no breakages caused by this change.

AlabasterAxe commented:
Okay, I asked my resident git wizard and I think I've got it. Please let me know if there are any other requests!

AlabasterAxe commented:
Hey All! 

All the tests look like they're good. I think I've addressed the comments. Is there anything else that I need to do from my side?

Thanks!

AlabasterAxe commented:
Quick ping, please let me know if there's anything I can do from my side!

Thanks,

Matt

IgorMinar committed:
fixup! fix(compiler): Fix look up of entryComponents in AOT Summaries

IgorMinar reviewed (approved):
lgtm for build changes (after some tweaks and force pushing)

IgorMinar commented:
g3 is happy: https://test.corp.google.com/ui#id=OCL:212518015:BASE:212524565:1536702296828:586fda42

IgorMinar committed:
fixup! fix(compiler): Fix look up of entryComponents in AOT Summaries
